SecretSanta/test/SecretSanta.Api.Tests/UserControllerTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SecretSanta.Api.Controllers;
+using SecretSanta.Domain.Models;
+using Moq.AutoMock;
+using Moq;
+using SecretSanta.Domain.Services;
+
+namespace SecretSanta.Api.Tests
+{
+    [TestClass]
+    public class UserControllerTests
+    {
+        protected AutoMocker Mocker { get; private set; }
+        protected Mock<IUserService> MockUserService { get; private set; }
+        [TestInitialize]
+        public void SetUpMock()
+        {
+            Mocker = new AutoMocker();
+            MockUserService = Mocker.GetMock<IUserService>();
+        }
+
+        [TestMethod]
+        public void PostUser_ValidInput_Returns201WithUrlAndObjectCreated()
+        {
+            DTO.User user = new DTO.User {
+                FirstName = "Grant", LastName = "Woods"};
+
+            MockUserService.Setup(x => x.AddUser(It.IsAny<User>()))
+                .Callback((User u) => { u.Id = 1; });
+            var controller = new UserController(MockUserService.Object);
+
+            var result = (CreatedResult)controller.PostUser(user);
+
+            var returnedUser = (DTO.User)result.Value;
+
+            Assert.AreEqual<int?>(201, result.StatusCode);
+            Assert.AreNotEqual<int>(0, returnedUser.Id);
+            Assert.AreEqual<string>($"api/User/{returnedUser.Id}", result.Location);
+            Assert.AreEqual<string>("Grant", returnedUser.FirstName);
+            Assert.AreEqual<string>("Woods", returnedUser.LastName);
+            MockUserService.VerifyAll();
```

Comment:
Typically with AutoMocker you would let the AutoMocker instance verify all expectations on every mock that it is tracking. 
```suggestion
            Mocker.VerifyAll();
```